### ISSUE TITLE
togmak should fail if the tag specified on the git data source does not exist

### ISSUE THREAD

**Issue by jgangemi** [srevinsaju/togomak#28]
the following config should throw an error b/c the tag doesn't exist:
```
data git cicd {
  url   = "..."
  tag = "asdf"
  files = [
    "python/togomak/build.hcl", "python/togomak/docs.hcl", "python/togomak/publish.hcl"
  ]
}
```

output:
```
togomak root +build
INFO[0000] togomak (version=1.2.1)
INFO[0000] [provider=git] Still pulling git repo data.git... [0s elapsed]
INFO[0001] [provider=git] Completed pulling git repo data.git in [1s elapsed]
```

**Comment by jgangemi** [comment]
seems it will also continue if the branch or tag is empty.

**Comment by srevinsaju** [comment]
In the case if branch and tag is empty, it will fallback to the defaults.

**Comment by jgangemi** [comment]
isn’t there a difference between `nil` and an empty string? 

i feel this is related to #23 - if i didn’t declare a default, that would be `nil` and the defaults could be used. if i did declare `default` it should complain if the string is empty b/c it could be a mistake.

or you need a warning message about it along with clearer docs on how it works 🙂

**Comment by srevinsaju** [comment]
In Go, a `nil` string is `""`, so when the HCL code is converted into `""` by default. so basically, a `""` and `nil` does not make a difference in this case. Also, you can't have a branch or a tag named `""`

**Comment by srevinsaju** [comment]
#29 adds more documentation on how the ref, tag, branch works, and their order of precedence.

**Comment by srevinsaju** [comment]
Closing this as resolved in v1.2.2
